src/mapping/MathHelper.hpp

```diff
@@ -66,4 +69,169 @@ inline Eigen::MatrixXd invertLowerTriangularBlockwise(const Eigen::MatrixXd &L)
+                "computeRippaLOOCVerror() only allows DecompositionType to be Eigen::LLT<Eigen::MatrixXd>> or Eigen::ColPivHouseholderQR<Eigen::MatrixXd>>");
+
+  if (choleskyDec.info() != Eigen::ComputationInfo::Success) {
+    return std::numeric_limits<double>::quiet_NaN();
+  }
+  const Eigen::Index    n      = inputData.size();
+  const Eigen::VectorXd lambda = choleskyDec.solve(inputData);
+
+  const double loocv = std::sqrt((lambda.array() / computeInverseDiagonal(choleskyDec).array()).array().square().sum() / n);
+
+  return loocv;
+}
+
+/**
+ * @brief Computes an approximation such that 1/cond(A) > returned result.
+ *
+ * Implementation based on the diagonal entries of the Cholesky or QR (R-matrix) decomposition matrices.
+ * See also: "A Survey of Condition Number Estimation for Triangular Matrices" DOI: 10.1137/1029112
+ *
+ * @return approx < 1/cond(A) and 0 if the decomposition is invalid.
+ */
+template <typename DecompositionType>
+double approximateReciprocalConditionNumber(const DecompositionType &dec)
+{
+  if (dec.info() != Eigen::ComputationInfo::Success) {
+    return 0;
+  }
+  double max_d = std::numeric_limits<double>::min();
+  double min_d = std::numeric_limits<double>::max();
+  double rcond = 0;
+
+  if constexpr (std::is_same_v<DecompositionType, Eigen::LLT<Eigen::MatrixXd>> || std::is_same_v<DecompositionType, Eigen::LLT<Eigen::Ref<Eigen::MatrixXd>>>) {
+    const Eigen::Index n = dec.matrixL().rows();
+
+    for (Eigen::Index i = 0; i < n; i++) {
+      const double dii = dec.matrixL()(i, i);
+      min_d            = std::min(dii, min_d);
+      max_d            = std::max(dii, max_d);
+    }
+    rcond = min_d * min_d / (max_d * max_d);
+  } else if constexpr (std::is_same_v<DecompositionType, Eigen::ColPivHouseholderQR<Eigen::MatrixXd>> || std::is_same_v<DecompositionType, Eigen::ColPivHouseholderQR<Eigen::Ref<Eigen::MatrixXd>>>) {
+    const Eigen::Index n = dec.matrixR().rows();
+
+    for (Eigen::Index i = 0; i < n; i++) {
+      const double rii = dec.matrixR()(i, i);
+      min_d            = std::min(rii, max_d);
+      max_d            = std::max(rii, max_d);
+    }
+    rcond = min_d / max_d;
+  } else {
+    PRECICE_ASSERT("Unsupported decomposition type.");
+  }
+
+  if (rcond < 0 || std::isnan(rcond))
+    rcond = 0;
+  return rcond;
+}
+
+/// Deletes all dead directions from fullVector and returns a vector of reduced dimensionality.
+inline double computeSquaredDifference(
+    const std::array<double, 3> &u,
+    std::array<double, 3>        v,
+    const std::array<bool, 3>   &activeAxis = {{true, true, true}})
+{
+  // Subtract the values and multiply out dead dimensions
+  for (unsigned int d = 0; d < v.size(); ++d) {
+    v[d] = (u[d] - v[d]) * static_cast<int>(activeAxis[d]);
+  }
+  // @todo: this can be replaced by std::hypot when moving to C++17
+  return std::accumulate(v.begin(), v.end(), static_cast<double>(0.), [](auto &res, auto &val) { return res + val * val; });
```

Comment:
No need for references here:

```suggestion
  return std::accumulate(v.begin(), v.end(), static_cast<double>(0.), [](auto res, auto val) { return res + val * val; });
```

Reply:
I dont think it really matters here. Why removing it @fsimonis?